### PR TITLE
feat: display list of services

### LINKED ISSUE #33: List of services
As a Refugee/Volunteer I want to see list of all services so that can find the best one for me. See Epic #7

Old version - https://wehelp.live/services

Design of card (see Figma):
Screenshot 2022-04-25 at 22 23 53

### PR DESCRIPTION
**Pull request by amanuela97**
## List of services

- a list showing all the available services 
- closes #33
- [demo](https://wehelpdemo.netlify.app/services)

### CODE REVIEW

File: components/TServiceListItem.vue [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          }
+        })
+      } catch (error) {
+        throw error 
+      }
+    }
+  }
+}
+</script>
+
```
Comment (VOlekh) [inline components/TServiceListItem.vue:132]: I would suggest putting component into the services folder

File: pages/services.vue [hunk 2]
```diff
@@ -1,11 +1,61 @@
...
+      }
+    },
+    setServices(services) {
+      services.forEach(element => {
+          this.services.push(element)
+      });
+    },
+  },
+}
+</script>
```
Comment (VOlekh) [inline pages/services.vue:61]: Add empty line

Comment (VOlekh) [inline pages/services.vue:61]: As a user, I push items on a list and expect the URL to be opened. Is that in scope or out of scope?

**Comment by razbakov** [comment]
@amanuela97 can you please remove shadows

File: pages/services.vue (hunk 2 above)
Comment (amanuela97) [reply pages/services.vue:61]: not entirely sure. I made this based on the live version of the web, and it worked similarly

Comment (VOlekh) [reply pages/services.vue:60]: Then I think need to notify @Kateryna Vasylenko issue could be added to the backlog for next week if it is important, if not discarded or saved for later

**Review by harshbrahmxatriya — APPROVED** [review]

**Review by razbakov — CHANGES_REQUESTED** [review]

File: .babelrc [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+              "targets": {
+                "node": "current"
+              }
+            }
+          ]
+        ]
+      }
+    }
+  }
+
```
Comment (VOlekh) [inline .babelrc:17]: Add new line

Comment (VOlekh) [inline .babelrc:17]: Add new line

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  ],
+  plugins: ['prettier'],
+  // add your custom rules here
+  rules: {
+    'no-console': 'off',
+    'vue/no-v-html': 'off',
+    'vue/comment-directive': 0,
+  },
+  ignorePatterns: ['functions'],
+}
```
Comment (VOlekh) [inline .eslintrc.js:25]: Add new line

Comment (VOlekh) [inline .eslintrc.js:25]: Add new line

File: .prettierignore [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+build
+coverage
+node_modules
+*.log
+.nuxt
+nuxt.d.ts
+.output
+.env
+.vscode
+local
```
Comment (VOlekh) [inline .prettierignore:11]: New line required:)

Comment (VOlekh) [inline .prettierignore:11]: New line required:)

File: .prettierrc [hunk 6]
```diff
@@ -1,5 +1,9 @@
...
   "semi": false,
   "arrowParens": "always",
-  "singleQuote": true
-}
+  "singleQuote": true,
+  "useTabs": false,
+  "tabWidth": 2,
+  "endOfLine": "lf",
+  "trailingComma": "es5"
+}
```
Comment (VOlekh) [inline .prettierrc:9]: Googled for ne line at the end of the file and found: In VS Code you can set "files.insertFinalNewline": true and Prettier won't override this setting.

Comment (VOlekh) [inline .prettierrc:9]: Googled for ne line at the end of the file and found: In VS Code you can set "files.insertFinalNewline": true and Prettier won't override this setting.

File: components/TServiceListItem.vue [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        default:
+          color = 'bg-green-100 text-green-800'
+          break
+      }
+      return style + color
+    },
+  },
+  created() {
+    this.location.forEach(async (location) => {
+      const result = await getCities(location)
```
Comment (razbakov) [inline components/TServiceListItem.vue:111]: please find a way to receive this data from parent component and use this one only as a decoration component without API calls

**Review by razbakov — CHANGES_REQUESTED** [review]

File: pages/services.vue (hunk 2 above)
Comment (VOlekh) [reply pages/services.vue:37]: out of scope

Comment (VOlekh) [reply pages/services.vue:37]: out of scope

File: airtable.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  const config = useRuntimeConfig()
+  const options = {
+    headers: {
+      Authorization: `Bearer ${config.API_KEY}`,
+    },
+  }
+  const url = `https://api.airtable.com/v0/${config.BASE_ID}/Cities/${location}`
+  const response = await $fetch(url, options)
+  if (response) return response
+}
```
Comment (VOlekh) [inline airtable.js:45]: End of file with new line

Comment (VOlekh) [inline airtable.js:45]: End of file with new line

File: airtable.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+export const getServices = async () => {
+  const config = useRuntimeConfig()
+  const options = {
+    headers: {
+      Authorization: `Bearer ${config.API_KEY}`,
+    },
+  }
+  let offset = null
+  do {
+    const offSet = offset ? `offset=${offset}` : ''
```
Comment (VOlekh) [inline airtable.js:20]: What is offset doing?

Comment (VOlekh) [inline airtable.js:20]: What is offset doing?

File: airtable.js (hunk 8 above)
Comment (VOlekh) [inline airtable.js:45]: Why the file called airtable.js ?

Comment (VOlekh) [inline airtable.js:45]: Why the file called airtable.js ?

File: components/TServiceListItem.vue [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+      this.cities.push(result)
+    })
+  },
+}
+</script>
```
Comment (VOlekh) [inline components/TServiceListItem.vue:116]: would you move the component to a services folder?
Add line)

Comment (VOlekh) [inline components/TServiceListItem.vue:116]: would you move the component to a services folder?
Add line)

File: package.json [hunk 11]
```diff
@@ -4,17 +4,26 @@
...
     "@nuxt/postcss8": "^1.1.3",
     "@tailwindcss/typography": "^0.5.2",
-    "autoprefixer": "^10.4.0",
+    "autoprefixer": "^10.4.2",
     "nuxt3": "latest",
     "postcss": "^8.4.5",
     "tailwind-config-viewer": "^1.6.2",
     "tailwindcss": "^3.0.6",
-    "vue-gtag-next": "^1.14.0"
+    "vue-gtag-next": "^1.14.0",
```
Comment (VOlekh) [inline package.json:18]: Why do you want to collect google analitics?

Comment (VOlekh) [inline package.json:18]: Why do you want to collect google analitics?

File: airtable.js (hunk 8 above)
Comment (razbakov) [reply airtable.js:45]: that's an airtable service, i would also call it like this, but I would move it in subfolder

File: package.json [hunk 12]
```diff
@@ -4,17 +4,26 @@
...
+    "@nuxtjs/eslint-config": "^2.0.0",
+    "@nuxtjs/eslint-module": "^1.0.0",
+    "babel-eslint": "^10.0.1",
+    "eslint": "^6.1.0",
+    "eslint-config-prettier": "^6.10.0",
+    "eslint-plugin-nuxt": ">=0.4.2",
+    "eslint-plugin-prettier": "^3.1.2",
+    "prettier": "^1.19.1"
```
Comment (VOlekh) [inline package.json:26]: line 36 end of file with empty line

Comment (VOlekh) [inline package.json:26]: line 36 end of file with empty line

File: airtable.js (hunk 9 above)
Comment (razbakov) [reply airtable.js:20]: that's a parameter of the api, used for pagination

File: .eslintrc.js (hunk 4 above)
Comment (razbakov) [reply .eslintrc.js:25]: @VOlekh why you have all your comments doubled?

File: tailwind.config.js [hunk 13]
```diff
@@ -11,7 +11,11 @@ module.exports = {
...
   theme: {
-    extend: {},
+    extend: {
+      gridTemplateColumns: {
+        'auto-fit': 'repeat(auto-fit, minmax(400px, 1fr))',
+      },
+    },
   },
   plugins: [require('@tailwindcss/typography')],
 }
```
Comment (VOlekh) [inline tailwind.config.js:21]: EL :)

Comment (VOlekh) [inline tailwind.config.js:21]: EL :)

**Review by VOlekh — APPROVED** [review]
Just need to add empty lines. Have some general questions in the comments.

File: package.json (hunk 11 above)
Comment (amanuela97) [reply package.json:18]: the dependency was already there.

**Comment by razbakov** [comment]
@amanuela97 there are conflicts, please resolve

**Comment by razbakov** [comment]
@amanuela97 please click "resolve conversation" as soon as you fix it

**Review by razbakov — CHANGES_REQUESTED** [review]
please resolve the conflicts

File: pages/services.vue [hunk 14]
```diff
@@ -1,11 +1,35 @@
...
+        :type="service.fields.Type"
+        :services="service.fields.Services"
+        :phone="service.fields.Phone"
+        :rating="service.fields.Rating"
+      />
+    </ul>
+  </div>
 </template>
+<script>
+import TServiceListItem from '../components/services/TServiceListItem.vue'
```
Comment (razbakov) [inline pages/services.vue:19]: Nuxt automatically imports any components in your components/ directory (along with components that are registered by any modules you may be using).

https://v3.nuxtjs.org/guide/directory-structure/components

**Review by razbakov — CHANGES_REQUESTED** [review]

**Review by razbakov — APPROVED** [review]
